map.go

```diff
@@ -1545,12 +1557,16 @@ func newMapIterator(target *Map) *MapIterator {
 	}
 }
 
-// Next decodes the next key and value.
+// Next decodes the next key and value. If the iterator is created
```

Comment:
Nit: do you think it's necessary to call this out again? It's already on Drain after all.